packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -141,15 +146,17 @@ class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProv
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    _textDirection = Directionality.of(context);
-    _painter = _buildCupertinoScrollbarPainter();
+    _painter ??= _buildCupertinoScrollbarPainter(context);
+    _painter
+      ..textDirection = Directionality.of(context)
+      ..color = CupertinoDynamicColor.resolve(_kScrollbarColor, context);
```

Comment:
Need to update the painter's padding too, since it depends on MediaQuery.of(context).